### PR TITLE
Remove unnecessary CSS comments from compiled styles

### PR DESCRIPTION
**Pull request by aristath**
Trac ticket: https://core.trac.wordpress.org/ticket/56388

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tools/webpack/shared.js [hunk 1]
```diff
@@ -57,16 +57,21 @@ const baseConfig = ( env ) => {
...
-			.then( ( result ) => result.css );
-	}
-	return content;
+	return postcss( [
+		require( 'cssnano' )( {
+			preset: mode === 'production' ? 'default' : [
+				'default',
+				{
+					discardComments: {
+						removeAll: ! content.includes( 'Copyright' ) && ! content.includes( 'License' ),
```
Comment (aristath) [inline tools/webpack/shared.js:66]: This condition ensures that Copyright/License comments don't get removed.

**Comment by SergeyBiryukov** [comment]
This appears to modify some files in the `wp-includes/blocks` directory, causing [some tests to fail](https://github.com/WordPress/wordpress-develop/runs/7853110976?check_suite_focus=true) 🤔 
```
Ensure version-controlled files are not modified or deleted during building and cleaning
diff --git a/src/wp-includes/blocks/site-logo/style-rtl.css b/src/wp-includes/blocks/site-logo/style-rtl.css
index 9062fc..c4546 100644
--- a/src/wp-includes/blocks/site-logo/style-rtl.css
+++ b/src/wp-includes/blocks/site-logo/style-rtl.css
```

**Comment by aristath** [comment]
> This appears to modify some files in the `wp-includes/blocks` directory, causing [some tests to fail](https://github.com/WordPress/wordpress-develop/runs/7853110976?check_suite_focus=true) 🤔
> 
> ```
> Ensure version-controlled files are not modified or deleted during building and cleaning
> diff --git a/src/wp-includes/blocks/site-logo/style-rtl.css b/src/wp-includes/blocks/site-logo/style-rtl.css
> index 9062fc..c4546 100644
> --- a/src/wp-includes/blocks/site-logo/style-rtl.css
> +++ b/src/wp-includes/blocks/site-logo/style-rtl.css
> ```

Yes, that is expected. The patch modifies the compilation of block stylesheets to remove comments from them. Looking at the failing tests, the modified stylesheets are what they should be 👍

**Comment by SergeyBiryukov** [comment]
If I understand correctly, it might be possible to make the same change in Gutenberg, so that when these CSS files are backported to core as part of package updates, they would no longer be flagged as modified by this PR, and then it can be merged 🤔

**Comment by aristath** [comment]
Gutenberg PR: https://github.com/WordPress/gutenberg/pull/43177

**Review by felixarntz — APPROVED** [review]
@aristath This looks like a reasonable change to me, however I would not exactly call myself a Webpack expert.

@gziolo Would you mind reviewing this PR as well please? Pinging you since you authored or at least committed the original file. :)

**Review by SergeyBiryukov — APPROVED** [review]

**Comment by SergeyBiryukov** [comment]
Looks good to me 👍

**Comment by gziolo** [comment]
It doesn't change any behavior for production, so we can proceed with it 👍🏻

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/55193

**Pull request closed without merging**